plugin/trino-openlineage/src/test/java/io/trino/plugin/openlineage/MarquezServer.java

```diff
@@ -0,0 +1,122 @@
+    private static final String POSTGRES_USER;
+    private static final String POSTGRES_PASSWORD;
+    private static final String DEFAULT_VERSION;
+
+    private Network network;
+
+    static {
+        MARQUEZ_CONFIG = "/opt/marquez/marquez.test.yaml";
+        MARQUEZ_PORT = 7000;
+        MARQUEZ_ADMIN_PORT = 7001;
+        MARQUEZ_HEALTCHECK_API = "/ping";
+        POSTGRES_HOST = "postgres";
+        POSTGRES_DB = "marquez";
+        POSTGRES_PORT = 5432;
+        POSTGRES_USER = "marquez";
+        POSTGRES_PASSWORD = "marquez";
+        DEFAULT_VERSION = "0.46.0";
+    }
+
+    private final GenericContainer<?> dockerContainerAPI;
+    private final GenericContainer<?> dockerContainerPostgres;
+
+    public MarquezServer()
+    {
+        this(DEFAULT_VERSION);
+    }
+
+    public MarquezServer(String version)
+    {
+        network = Network.newNetwork();
+        this.dockerContainerPostgres = new GenericContainer<>("postgres:14")
+                .withNetwork(network)
+                .withNetworkAliases(POSTGRES_HOST)
+                .withEnv("POSTGRES_USER", POSTGRES_USER)
+                .withEnv("POSTGRES_PASSWORD", POSTGRES_PASSWORD)
+                .withEnv("POSTGRES_DB", POSTGRES_DB)
+                .withStartupTimeout(Duration.ofSeconds(360));
+
+        this.dockerContainerPostgres.start();
+
+        this.dockerContainerAPI = new GenericContainer<>("marquezproject/marquez:" + version)
+                .withNetwork(network)
+                .withExposedPorts(MARQUEZ_PORT, MARQUEZ_ADMIN_PORT)
+                .withEnv("MARQUEZ_PORT", String.valueOf(MARQUEZ_PORT))
+                .withEnv("MARQUEZ_ADMIN_PORT", String.valueOf(MARQUEZ_ADMIN_PORT))
+                .withEnv("POSTGRES_URL", getPostgresUri())
+                .withEnv("POSTGRES_USER", "marquez")
+                .withEnv("POSTGRES_PASSWORD", "marquez")
+                .withEnv("MARQUEZ_CONFIG", MARQUEZ_CONFIG)
+                .withCopyFileToContainer(
+                        MountableFile.forClasspathResource("marquez.yaml"),
+                        MARQUEZ_CONFIG)
+                .withCommand("/opt/marquez/wait-for-it.sh", format("%s:%s", POSTGRES_HOST, POSTGRES_PORT), "--", "./entrypoint.sh")
+                .waitingFor(Wait.forHttp(MARQUEZ_HEALTCHECK_API)
+                        .forPort(MARQUEZ_ADMIN_PORT)
+                        .forStatusCode(200))
+                .withStartupTimeout(Duration.ofSeconds(360));
+
+        this.dockerContainerAPI.start();
+    }
+
+    public String getPostgresUri()
+    {
+        return format("jdbc:postgresql://%s:%s/%s", POSTGRES_HOST, POSTGRES_PORT, POSTGRES_DB);
+    }
+
+    public URI getMarquezUri()
+    {
+        return URI.create("http://" + dockerContainerAPI.getHost() + ":" + dockerContainerAPI.getMappedPort(MARQUEZ_PORT));
+    }
+
+    @Override
+    public void close()
+    {
+        dockerContainerPostgres.close();
+        dockerContainerAPI.close();
+        network.close();
```

Comment:
We could use `Closer` , register them closing logic for the container in the container and we could use `closer.close()` here

Reply:
great idea, done